### ISSUE TITLE
Forward after authentication attempt configuration support

### ISSUE THREAD

**Issue by rwinch** [spring-projects/spring-security#3728]
We should provide first class configuration support for forwarding after authentication attempts (both success / failure). For example, we might have the following:

```
formLogin()
    .failureForwardUrl("/failure")
```

In XML configuration something like the following would be provided:

```
<form-login authentication-failure-forward-url="/failure"/>
```

Something similar would be provided for authentication success.

**Comment by shazin** [comment]
@rwinch Since there are already `authentication-failure-url` in the `form-login`, I suggest the following two names 

`authentication-failure-forward-url`, `authentication-success-forward-url` 

This will distinguish that those are for forward.

**Comment by rwinch** [comment]
@shazin Thanks I had the intention of using `authentication-failure-forward-url` in the example and somehow messed it up. This is now corrected.

**Comment by shazin** [comment]
Solved in pull request #3740
